test/built-ins/ArrayBuffer/prototype/sliceToImmutable/argument-coercion.js

```diff
@@ -0,0 +1,369 @@
+  5. If end is undefined, let relativeEnd be len; else let relativeEnd be ? ToIntegerOrInfinity(end).
+  6. If relativeEnd = -∞, let to be 0.
+  7. Else if relativeEnd < 0, let to be max(len + relativeEnd, 0).
+  8. Else, let to be min(relativeEnd, len).
+  9. Return the Record { [[From]]: from, [[To]]: to }.
+
+  ToIntegerOrInfinity ( argument )
+  1. Let number be ? ToNumber(argument).
+  2. If number is one of NaN, +0𝔽, or -0𝔽, return 0.
+  3. If number is +∞𝔽, return +∞.
+  4. If number is -∞𝔽, return -∞.
+  5. Return truncate(ℝ(number)).
+features: [immutable-arraybuffer]
+includes: [compareArray.js]
+---*/
+
+var ab = new ArrayBuffer(8);
+assert.sameValue(ab.sliceToImmutable().byteLength, 8,
+  "Must default to receiver byteLength.");
+ab = new ArrayBuffer(8);
+assert.sameValue(ab.sliceToImmutable(undefined, undefined).byteLength, 8,
+  "Must default (undefined, undefined) to receiver byteLength.");
+
+function repr(value) {
+  if (typeof value === "string") return JSON.stringify(value);
+  if (typeof value === "bigint") return String(value) + "n";
+  if (!value && 1/value === -Infinity) return "-0";
+  return String(value);
+}
+
+var make32ByteArrayBuffer() {
+  var ab = new ArrayBuffer(32);
+  var view = new Uint8Array(ab);
+  for (var i = 0; i < 8; i++) view[i] = i + 1;
+  return ab;
+}
+
+var goodInputs = [
+  // Unmodified non-negative integral numbers
+  [0, 0],
+  [1, 1],
+  [10, 10],
+  // Truncated non-negative integral numbers
+  [0.9, 0],
+  [1.9, 1],
+  [-0.9, 0],
+  // Negative integral numbers
+  [-1, -1],
+  [-1.9, -1],
+  [-2.9, -2],
+  // Coerced to integral numbers
+  [-0, 0],
+  [null, 0],
+  [false, 0],
+  [true, 1],
+  ["", 0],
+  ["8", 8],
+  ["+9", 9],
+  ["-9", -9],
+  ["10e0", 10],
+  ["+1.1E+1", 11],
+  ["+.12e2", 12],
+  ["130e-1", 13],
+  ["0b1110", 14],
+  ["0XF", 15],
+  ["0xf", 15],
+  ["0o20", 16],
+  // Coerced to NaN and mapped to 0
+  [NaN, 0],
+  ["7up", 0],
+  ["1_0", 0],
+  ["0x00_ff", 0],
+  // Clamped
+  [-32, 0],
+  ["-32", 0],
+  [-Infinity, 0],
+  ["-Infinity", 0],
+  [33, 32],
+  ["33", 32],
+  [Infinity, 32],
+  ["Infinity", 32]
+];
+
+for (var i = 0; i < goodInputs.length; i++) {
+  var rawStart = goodInputs[i][0];
+  var intStart = goodInputs[i][1];
+  for (var j = 0; j < goodInputs.length; j++) {
+    var rawEnd = goodInputs[j][0];
+    var intEnd = goodInputs[j][1];
+    var intLength = intEnd - intStart;
+    var source = make32ByteArrayBuffer();
+    var expectContents = Array.from(new Uint8Array(source)).slice(rawStart, rawEnd);
+    var dest = source.sliceToImmutable(rawStart, rawEnd);
+    assert.sameValue(dest.byteLength, intLength,
+      "sliceToImmutable(" + repr(rawStart) + ", " + repr(rawEnd) + ")");
+    assert.compareArray(new Uint8Array(dest), expectContents,
+      "sliceToImmutable(" + repr(rawStart) + ", " + repr(rawEnd) + ")");
+    assert.sameValue(dest.immutable, true,
+      "sliceToImmutable(" + repr(rawStart) + ", " + repr(rawEnd) + ")");
```

Comment:
Minor suggestion, add something to these assertion messages to distinguish which assertion is failing (byte length, contents, or immutability) (Same for below)